File: app/src/main/java/com/lumiyaviewer/lumiya/voice/common/messages/VoiceEnableMic.java
package com.lumiyaviewer.lumiya.voice.common.messages;

import android.os.Bundle;
import com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage;

public class VoiceEnableMic implements VoicePluginMessage {
    public final boolean enableMic;

    public VoiceEnableMic(Bundle bundle) {
        this.enableMic = bundle.getBoolean("enableMic");
    }

    public VoiceEnableMic(boolean enableMic) {
        this.enableMic = enableMic;
    }

    @Override
    public Bundle toBundle() {
        Bundle bundle = new Bundle();
        bundle.putBoolean("enableMic", this.enableMic);
        return bundle;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/voice/common/messages/VoiceEnableMic.kt
package com.lumiyaviewer.lumiya.voice.common.messages

import android.os.Bundle
import com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage

class VoiceEnableMic : VoicePluginMessage {
    @JvmField
    val enableMic: Boolean

    constructor(bundle: Bundle) {
        this.enableMic = bundle.getBoolean("enableMic")
    }

    constructor(enableMic: Boolean) {
        this.enableMic = enableMic
    }

    override fun toBundle(): Bundle = Bundle().apply {
        putBoolean("enableMic", enableMic)
    }
}
